picard/log.py

```diff
@@ -55,20 +55,22 @@ def debug_mode(enabled):
 
 class TailLogHandler(logging.Handler):
 
-    def __init__(self, log_queue, tail_logger):
+    def __init__(self, log_queue, tail_logger, log_queue_lock):
         super().__init__()
         self.log_queue = log_queue
         self.tail_logger = tail_logger
+        self.log_queue_lock = log_queue_lock
         self.pos = 0
 
     def emit(self, record):
-        self.log_queue.append(
-            TailLogTuple(
-                self.pos,
-                self.format(record),
-                record.levelno
+        with self.log_queue_lock:
+            self.log_queue.append(
```

Comment:
It's not quite clear to me why this is necessary. According to [the documentation](https://docs.python.org/3.6/library/collections.html#collections.deque), " Deques support thread-safe, memory efficient appends and pops from either side of the deque". Why is it required to protect `append` with a lock?

Reply:
The main problem was https://community.metabrainz.org/t/debug-logging-in-picard-2-0/391297/5?u=samj1912

Basically if the deque is changed while iterating, which happens when there are too many log events.

Reply:
Ah, makes sense.

I think a crash like that deserves a JIRA ticket, though.